src/tools/compiletest/src/header/needs.rs

```diff
@@ -171,9 +171,12 @@ pub(super) fn handle_needs(
         },
     ];
 
-    let (name, comment) = match ln.split_once([':', ' ']) {
-        Some((name, comment)) => (name, Some(comment)),
-        None => (ln, None),
+    // Because `needs-target-has-atomic` accepts comma separated arguments after a colon to specify
+    // data sizes, we check whether comment starts with colon.
+    let (name, comment, comment_starts_with_colon) = if let Some(index) = ln.find([':', ' ']) {
+        (&ln[..index], Some(&ln[index + 1..]), ln.as_bytes()[index] == b':')
+    } else {
+        (ln, None, false)
     };
```

Comment:
Suggestion: this also accepts a whitespace separator, please only accept `:` to help consistency.

Reply:
This accepts a whitespace separator, but only when the directive isn't `needs-target-has-atomic`. See [here](https://github.com/kei519/rust/blob/a74764c83ca2de1fe281a279ab9be612900fe074/src/tools/compiletest/src/header/needs.rs#L194).
It should accept a whitespace separator when other than `needs-target-has-atomic` because it is currently accepted. Shouldn't it?

Reply:
I'll double-check tmrw